core/api/views/business_plan.py

```diff
@@ -160,8 +179,22 @@ def update(self, request, *args, **kwargs):
         self.perform_create(serializer)
         new_instance = serializer.instance
 
+        # check user permissions
+        user = request.user
+        if (
+            user.agency_role == user.AgencyRole.AGENCY_INPUTTER
+            and new_instance.status != BusinessPlan.Status.draft
```

Comment:
Here the new_instance.status should be always draft because the status change from draft will be made with another request for another endpoint.
You can check that now or you can add a comment to be sure we will update this part when the status-update endpoint will be created